### ISSUE TITLE
Fatal error:  Uncaught Failed to invoke method add in class com.infusionsoft.api.xmlrpc.XmlRpcDataService: null

### ISSUE THREAD

**Issue by alantoday** [novaksolutions/infusionsoft-php-sdk#60]
Hi, 
I setup and got everything working and tested out okay - thanks for creating the tool kit.
I was able to add, update and delete credit cards fine for months. Suddenly today I get an
error when I try to add a new card and the "null" in the error message (and error message
in general) isn't really helping me. Hoping it makes more sense to you and you provide
a suggestion/fix.

The error message on the screen is:

Fatal error:  Uncaught Failed to invoke method add in class com.infusionsoft.api.xmlrpc.XmlRpcDataService: null
Attempted: 1 time(s).
  thrown in /home/digital/public_html/da/scripts/Infusionsoft/App.php on line 172

This is my code - when I inspect the message before sending - all the data appear to be there and valid.

	$credit_card = new Infusionsoft_CreditCard();
	$credit_card->ContactId = $inf_contact_id;
	$credit_card->CardNumber = $CardNumber;
	$credit_card->CVV2 = $CVV2;
	$credit_card->CardType = WriteCardType($CardNumber);
	$credit_card->Status = 0; //0: Unknown, 1: Invalid, 2: Deleted, 3: Valid/Good, 4: Inactive
	$credit_card->ExpirationMonth = $ExpirationMonth;
	$credit_card->ExpirationYear = $ExpirationYear;
	$credit_card->BillName = $BillName;
	$credit_card->BillAddress1 = $BillAddress1;
	$credit_card->BillCity = $BillCity;
	$credit_card->BillState = $BillState;
	$credit_card->BillZip = $BillZip;
	$credit_card->BillCountry = $BillCountry;
	return $credit_card->save();

This is the $result I get:

HTTP/1.1 100 Continue

HTTP/1.1 200 OK
Server: Apache-Coyote/1.1
Pragma: no-cache
Cache-Control: no-cache, no-store
Expires: Fri, 08 Apr 2016 07:00:53 GMT
X-FRAME-OPTIONS: SAMEORIGIN
Content-Type: text/xml;charset=UTF-8
Content-Length: 326
Date: Fri, 08 Apr 2016 07:00:53 GMT
Set-Cookie: app-lb=!U7CZu+/CYJYIAADrjfnY7K/Z+Br72+gwNeNBUHgfdOu2u9qlQfP69GmweiBWt448wDbeHHr0RMWAZ19de8Z3LxC9JEyHSQHrviDFkOdRczCAE56A3lS+B5Vp9IdzAj1ByWaz7l+3AHRRhLQfVlCFarMRsHIb4xg=; path=/

faultCode0faultStringFailed to invoke method add in class com.infusionsoft.api.xmlrpc.XmlRpcDataService: null

**Comment by alantoday** [comment]
I should add - that the credit card actually does get added (so at least that's good - in a way) - but the fatal error isn't very reassuring to the user.

**Comment by joeynovak** [comment]
Hey Alan,

Thanks for reporting this.  This error is because Infusionsoft is reporting "Failed to invoke method add in class com.infusionsoft.api.xmlrpc.XmlRpcDataService: null"  we pass back the error from Infusionsoft when the sdk gets a response it doesn't know how to handle.  I don't know the cause of this error.  Usually, Infusionsoft gives us more information where in your error it says "null".

Because it is saying "null" and not what is really wrong, I think it is a problem on their end.  Also, no one else has reported this problem yet, so it may even just be your app.

Can you please keep us updated on what you find out?  Does it go away, does Infuionsoft fix it? etc...

Thanks!

Joey

**Comment by alantoday** [comment]
Hi Joey,
I talked with Infusionsoft and they said:
-------
It appears to have to deal with adding a credit card without a NameOnCard being passed to Infusionsoft. We are hoping to get a fix in early next week. Until then passing the NameOnCard should get you past this issue.
-------
Sure enough if I passed in NameOnCard instead of BillName (which probably makes more sense anyway) I no longer got the error.
